### PR TITLE
refactor: Make tests generic w.r.t. curve/group

### PR DESCRIPTION
**Pull request by oskarth**
Addresses https://github.com/privacy-scaling-explorations/Nova/issues/34

Should cover all files

### CODE REVIEW

File: src/ccs/mod.rs [hunk 1]
```diff
@@ -471,12 +469,12 @@ pub mod test {
 
     // generate generators and ro constants
     let _ck = S.commitment_key();
-    let _ro_consts =
-      <<G as Group>::RO as ROTrait<<G as Group>::Base, <G as Group>::Scalar>>::Constants::new();
+    let _ro_consts = <G::RO as ROTrait<G::Base, G::Scalar>>::Constants::new();
+    //      <<G as Group>::RO as ROTrait<<G as Group>::Base, <G as Group>::Scalar>>::Constants::new();
```
Comment (arnaucube) [inline src/ccs/mod.rs:473]: Is this commented code intended to be left here?

File: src/ccs/cccs.rs [hunk 2]
```diff
@@ -158,17 +158,14 @@ mod tests {
     vecs.iter().map(Vec::as_slice).collect()
   }
 
-  /// Do some sanity checks on q(x). It's a multivariable polynomial and it should evaluate to zero inside the
-  /// hypercube, but to not-zero outside the hypercube.
```
Comment (CPerezz) [inline src/ccs/cccs.rs:162]: Why is the comment removed?

**Review by CPerezz — CHANGES_REQUESTED** [review]
I'm not in favour of merging this. 
It's non-sensical to actually transform the tests into trait-dependant fns and then create new tests that call the old ones passing the type of the generic. 

It just adds more code, and no apparent benefit. As whatever passes the tests for pasta, will do for any other cycle.

**Comment by arnaucube** [comment]
Regarding generic tests, looks like is also being done in upstream, eg: https://github.com/microsoft/Nova/blob/main/src/nifs.rs#L388

File: src/ccs/cccs.rs (hunk 2 above)
Comment (arnaucube) [reply src/ccs/cccs.rs:162]: It has been moved some lines down, on top of the generic test invoker.

Comment (CPerezz) [reply src/ccs/cccs.rs:162]: Just saw yes

**Review by CPerezz — APPROVED** [review]
After discussing privately, seems I agreed to go that way :sweat_smile: hahahah

So let's go that way considering that we also have the same format upstream.
Apologies for the noise.

File: src/ccs/mod.rs (hunk 1 above)
Comment (oskarth) [reply src/ccs/mod.rs]: Accidents, thanks! Fixed in https://github.com/privacy-scaling-explorations/Nova/pull/40/commits/bc5aa34dd02413b1e2886ec36e0fbcbdbb2424ef

**Review by arnaucube — APPROVED** [review]

**Comment by oskarth** [comment]
Ready to be merged

**Comment by oskarth** [comment]
CI passed before https://github.com/privacy-scaling-explorations/Nova/actions/runs/5577103414 and only comment change, merging now to make it easier for others to rebase

**Pull request merged**